cmd/microcloud/waitready.go

```diff
@@ -31,10 +32,17 @@ func (c *cmdWaitready) Run(cmd *cobra.Command, args []string) error {
 	}
 
 	options := microcluster.Args{StateDir: c.common.FlagMicroCloudDir, Verbose: c.common.FlagLogVerbose, Debug: c.common.FlagLogDebug}
-	m, err := microcluster.App(context.Background(), options)
+	m, err := microcluster.App(options)
 	if err != nil {
 		return err
 	}
 
-	return m.Ready(c.flagTimeout)
+	ctx := context.Background()
+	if c.flagTimeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(c.flagTimeout))
+		defer cancel()
+	}
+
+	return m.Ready(ctx)
```

Comment:
Ok I've just seen these commits related to passing contexts around are related to a change to microcluster. Can we move these to a separate PR for updating dependencies? Then this PR can focus just on adding the proxy helper.

Reply:
Sure, although in the interest of not growing the number of PRs that are unable to be merged, I'll hold off on that until it's safe to update microcloud's dependency :)